menu/drivers/ozone.c

```diff
@@ -1249,15 +1251,29 @@ static unsigned ozone_entries_icon_get_id(ozone_handle_t *ozone,
 #endif
 
    if (
-         (type >= MENU_SETTINGS_INPUT_DESC_BEGIN) &&
+         (type >= MENU_SETTINGS_INPUT_BEGIN) &&
          (type <= MENU_SETTINGS_INPUT_DESC_END)
       )
       {
          unsigned input_id;
-         input_id = MENU_SETTINGS_INPUT_DESC_BEGIN;
-         while (type > (input_id + 23))
+         if (type < MENU_SETTINGS_INPUT_DESC_BEGIN)
+         {
+            input_id = MENU_SETTINGS_INPUT_BEGIN;
+            if ( type == input_id + 2)
+               return OZONE_ENTRIES_ICONS_TEXTURE_INPUT_SETTINGS;
+            if ( type == input_id + 4)
+               return OZONE_ENTRIES_ICONS_TEXTURE_RELOAD;
+            if ( type == input_id + 5)
+               return OZONE_ENTRIES_ICONS_TEXTURE_SAVING;
+            input_id = input_id + 7;
+         }
+         else
          {
-            input_id = (input_id + 24) ;
+            input_id = MENU_SETTINGS_INPUT_DESC_BEGIN;
+            while (type > (input_id + 23))
+            {
+               input_id = (input_id + 24) ;
```

Comment:
`2`, `4`, `5`, `7`, `23`, `24` : what are those values? Where do they come from?

Reply:
Can this be cleared up/made less opaque later on? Maybe by adding a few comments explaining this?